timeseries/tests/unittests/models/test_gluonts.py

```diff
@@ -203,3 +210,52 @@ def test_when_tft_quantiles_are_deciles_then_forecast_contains_correct_quantiles
     predictions = model.predict(data=DUMMY_TS_DATAFRAME)
     assert "mean" in predictions.columns
     assert all(str(q) in predictions.columns for q in quantiles)
+
+
+@pytest.fixture(scope="module")
+def df_with_static():
+    feature_pipeline = ContinuousAndCategoricalFeatureGenerator(verbosity=0)
+    df = DUMMY_VARIABLE_LENGTH_TS_DATAFRAME_WITH_STATIC.copy(deep=False)
+    df.static_features = feature_pipeline.fit_transform(df.static_features)
+    return df
+
+
+@pytest.mark.parametrize("model_class", MODELS_WITH_STATIC_FEATURES)
+def test_when_static_features_present_then_they_are_passed_to_dataset(model_class, df_with_static):
+    model = model_class()
+    with mock.patch(
+        "autogluon.timeseries.models.gluonts.abstract_gluonts.SimpleGluonTSDataset.__init__"
+    ) as patch_dataset:
+        try:
+            model.fit(train_data=df_with_static)
+        except TypeError:
+            call_kwargs = patch_dataset.call_args[1]
+            feat_static_cat = call_kwargs["feat_static_cat"]
+            feat_static_real = call_kwargs["feat_static_real"]
+            assert (feat_static_cat.dtypes == "category").all()
+            assert (feat_static_real.dtypes == "float").all()
+
+
+@pytest.mark.parametrize("model_class", MODELS_WITH_STATIC_FEATURES)
+def test_when_static_features_present_then_model_attributes_set_correctly(model_class, df_with_static):
+    model = model_class(hyperparameters={"epochs": 1, "num_batches_per_epoch": 1})
+    model.fit(train_data=df_with_static)
+    assert model.use_feat_static_cat
+    assert model.use_feat_static_real
+    assert len(model.feat_static_cat_cardinality) == 1
```

Comment:
should we assert the content of cardinality as well?

Reply:
The static features are generated randomly at test time, we don't know it in advance. Should we make the generator deterministic?

Reply:
I think we might. at least randomly with fixed cardinality for both cardinality = 1 and many?

Reply:
How about now?